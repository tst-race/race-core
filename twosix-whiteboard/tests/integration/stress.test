#!/bin/bash

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
source "${DIR}"/helpers.sh

start_containers

# 10 megabyte string
LONG_STRING=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c 10M)

# TODO health check?
sleep 2

echo "post huge strings" | tee >(cat >&2)
(
for (( i = 0; i < 100; i++ )); do
    (
    curl --fail --silent --show-error --request POST localhost:5000/post/long --header 'Content-Type: application/json' -d @- <<CURL_DATA | jq -c  'del(.timestamp)'
{ "data": "$LONG_STRING" }
CURL_DATA
    ) &
done
) | sort

wait

echo "get index huge string (shouldn't exist becaue resizing)" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/0" | wc

echo "get index huge string" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/-1" | wc

echo "get range huge strings" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/0/-1" | wc

stop_containers