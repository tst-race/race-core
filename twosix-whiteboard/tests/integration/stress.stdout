building docker image in case it's not up-to-date
create docker network if in case it doesn't exist
start containers
post huge strings
{"index":0}
{"index":1}
{"index":10}
{"index":11}
{"index":12}
{"index":13}
{"index":14}
{"index":15}
{"index":16}
{"index":17}
{"index":18}
{"index":19}
{"index":2}
{"index":20}
{"index":21}
{"index":22}
{"index":23}
{"index":24}
{"index":25}
{"index":26}
{"index":27}
{"index":28}
{"index":29}
{"index":3}
{"index":30}
{"index":31}
{"index":32}
{"index":33}
{"index":34}
{"index":35}
{"index":36}
{"index":37}
{"index":38}
{"index":39}
{"index":4}
{"index":40}
{"index":41}
{"index":42}
{"index":43}
{"index":44}
{"index":45}
{"index":46}
{"index":47}
{"index":48}
{"index":49}
{"index":5}
{"index":50}
{"index":51}
{"index":52}
{"index":53}
{"index":54}
{"index":55}
{"index":56}
{"index":57}
{"index":58}
{"index":59}
{"index":6}
{"index":60}
{"index":61}
{"index":62}
{"index":63}
{"index":64}
{"index":65}
{"index":66}
{"index":67}
{"index":68}
{"index":69}
{"index":7}
{"index":70}
{"index":71}
{"index":72}
{"index":73}
{"index":74}
{"index":75}
{"index":76}
{"index":77}
{"index":78}
{"index":79}
{"index":8}
{"index":80}
{"index":81}
{"index":82}
{"index":83}
{"index":84}
{"index":85}
{"index":86}
{"index":87}
{"index":88}
{"index":89}
{"index":9}
{"index":90}
{"index":91}
{"index":92}
{"index":93}
{"index":94}
{"index":95}
{"index":96}
{"index":97}
{"index":98}
{"index":99}
get index huge string (shouldn't exist becaue resizing)
      5      34     233
get index huge string
      1       4 10485807
get range huge strings
      1      52 492830968
stop containers
